pkg/util/cluster/cluster.go

```diff
@@ -252,12 +261,36 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 		if err != nil {
 			return err
 		}
+
+		c.log.Info("peering subnets to CI infra")
+		err = c.peerSubnetsToCI(ctx, vnetResourceGroup, clusterName)
```

Comment:
Looking at https://ms.portal.azure.com/#@microsoft.onmicrosoft.com/resource/subscriptions/46626fc5-476d-41ad-8c76-2ec49c6994eb/resourceGroups/e2einfra-eastus/providers/Microsoft.Network/virtualNetworks/dev-vnet/peerings I think there's going to need to be an (albeit best effort) deletion cleanup step

Reply:
done